services/venice-router/src/main/java/com/linkedin/venice/router/VeniceRouterConfig.java

```diff
@@ -393,6 +400,12 @@ private void checkProperties(VeniceProperties props) {
     perStoreRouterQuotaBuffer = props.getDouble(ROUTER_PER_STORE_ROUTER_QUOTA_BUFFER, 1.5);
     httpClientOpensslEnabled = props.getBoolean(ROUTER_HTTP_CLIENT_OPENSSL_ENABLED, true);
     identityParserClassName = props.getString(IDENTITY_PARSER_CLASS, DefaultIdentityParser.class.getName());
+    singleGetLongTailRetryBudgetPercentDecimal =
+        props.getDouble(ROUTER_SINGLE_GET_LONG_TAIL_RETRY_BUDGET_PERCENT_DECIMAL, 0.1);
```

Comment:
Are these the default for fast client ? Do you think we should have higher limit for router ? Is it possible to make it dependent on actual QPS being used, like for 10% for 1M qps, but 50% for 1k QPS ?

Reply:
10% is the default for fast client. What is the logic behind a higher retry budget for lower KPS especially with store level isolation? There might be un-utilized resource when overall KPS is low but in such scenario we should expect less retries. Based on today's metrics the amount of long tail retry is relatively small in a healthy cluster.

Reply:
I was thinking compared to overall KPS in the router. eg ie for stores with 1k KPS and 1M qps, both should not be doing same 10% retries ?

Reply:
I'm still not sure if I fully understand the concern here. The 10% retry quota is the default limit of how much we are willing to perform long tail retry based on the user's long tail retry threshold config. It has nothing to do how much we plan/project to perform retry because we don't have full control of that. If this 10% quota is exhausted either:
1. User's long tail retry threshold is too aggressive (they can set it to 0ms and it becomes speculative query).
2. We have issues in our servers (due to regression or under provisioned).

Setting different quota for different workload without sufficient supporting data will just complicate the behavior and potentially mask the two signals above.

Reply:
Discussed offline with Xun, we could have the current setup for now and see if in future we would need per store configs to control different retry behavior.